docs/src/test-parameterize-js.md

```diff
@@ -9,13 +9,42 @@ You can either parameterize tests on a test level or on a project level.
 ## Parameterized Tests
 
 ```js title="example.spec.ts"
-const people = ['Alice', 'Bob'];
-for (const name of people) {
-  test(`testing with ${name}`, async () => {
-    // ...
-  });
+[
+  { name: 'Alice', expected: 'Hello, Alice!' },
+  { name: 'Bob', expected: 'Hello, Bob!' },
+  { name: 'Charlie', expected: 'Hello, Charlie!' },
+].forEach(({ name, expected }) => {
   // You can also do it with test.describe() or with multiple tests as long the test name is unique.
-}
+  test(`testing with ${name}`, async ({ page }) => {
+    await page.goto(`/greet?name=${name}`);
+    await expect(page.getByRole('heading', { name })).toBeVisible();
+  });
+});
+```
+
+### Before and after hooks
+
+When looping over `before{Each,All}` or `after{Each,All}` hooks, they will be executed for every iteration, so in most cases, you should loop only over tests:
```

Comment:
```suggestion
Most of the time you should put `beforeEach`, `beforeAll`, `afterEach` and `afterAll` hooks outside of `forEach`, so that hooks are executed just once:
```